### PR TITLE
client,cmd/snap: clarify name ambiguity in Plug or Slot

### PR DESCRIPTION
**Pull request by zyga**
This patch changes the public snappy API to refer to Plug.Name as .Plug
and Slot.Name as .Slot.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

**Comment by niemeyer** [comment]
Hmm.. that doesn't look like an improvement to me. The reasoning for using it at the API edge is to reinforce the terminology and make the data more obvious. Inside out own API, having Plug.Plug will likely not help much, comparing it to Plug.Plug.

**Comment by zyga** [comment]
I'm closing this temporarily. After discussing this online we agreed to use mixed json/go field naming. For Go we'll have Plug.Name and Slot.Name but in JSON we'll have {"plug": "..."} and {"slot": "..."} to differentiate the two fields.

File: cmd/snap/cmd_interfaces.go [hunk 1]
```diff
@@ -68,18 +68,18 @@ func (x *cmdInterfaces) Execute(args []string) error {
 			if x.Positionals.Query.Snap != "" && x.Positionals.Query.Snap != plug.Snap {
 				continue
 			}
-			if x.Positionals.Query.Name != "" && x.Positionals.Query.Name != plug.Name {
+			if x.Positionals.Query.Name != "" && x.Positionals.Query.Name != plug.Plug.Name {
```
Comment (stevenwilkin) [inline cmd/snap/cmd_interfaces.go]: Would `connection.Plug.Name` or similar make more sense here? I see that `AllPlugs()` returns `[]PlugConnections`

Comment (zyga) [inline cmd/snap/cmd_interfaces.go]: This is fixed with the upcoming /2.0/interfaces API change. It actually returns ``InterfaceConnections`` then but you'll have to wait a sec to see it.

**Comment by stevenwilkin** [comment]
:+1:

**Pull request merged**